src/org/zaproxy/zap/extension/websocket/WebSocketProxy.java

```diff
@@ -1001,4 +1012,52 @@ public boolean isAllowAPI() {
 	public void setAllowAPI(boolean allowAPI) {
 		this.allowAPI = allowAPI;
 	}
+	
+	/**
+	 * Return an instance of {@link HandshakeConfig} which is used to establish a new WebSocket Connection
+	 * see {@link ServerConnectionEstablisher#send(HandshakeConfig)} and {@link WebSocketProxy#reEstablishConnection(HandshakeConfig)}
+	 * The {@link HandshakeConfig} includes the HttpHandshake of the current connection. In addition, the method adds the
+	 * {@link WebSocketObserver} and {@link WebSocketSenderListener}  of the current connection to HandshakeConfig
+	 *
+	 * @return the HandshakeConfig
+	 * @throws IllegalArgumentException if an error occurred while trying to retrieve the HTTP handshake from history
+	 */
+	public HandshakeConfig getHandShakeConfig() throws IllegalArgumentException {
+		HandshakeConfig handshakeConfig;
+		try {
+			handshakeConfig = new HandshakeConfig(new HttpMessage(handshakeReference.getHttpMessage().getRequestHeader()));
+			for(WebSocketObserver webSocketObserver : observerList){
+				handshakeConfig.addChannelObserver(webSocketObserver);
+			}
+			
+			for(WebSocketSenderListener senderListener : senderListenerList ){
+				handshakeConfig.addChannelSenderListener(senderListener);
+			}
+		} catch (Exception e) {
+			throw new IllegalArgumentException("An error occurred while trying to retrieve the HTTP handshake from history ", e);
```

Comment:
@thc202 I used `IllegalArgumentException`. I am not sure that's the most suitable.